src/celeritas/optical/detail/CerenkovPreGenExecutor.hh

```diff
@@ -59,8 +57,9 @@ CerenkovPreGenExecutor::operator()(CoreTrackView const& track)
     cerenkov_dist = {};
 
     auto sim = track.make_sim_view();
-    auto optmat_id = get_optical_material(track);
-    if (!optmat_id || sim.step_length() == 0)
+    auto step = state.step[tsid];
```

Comment:
```suggestion
    auto const& step = state.step[tsid];
```

And I also noticed that `CerenkovPreGenerator` and `ScintillationPreGenerator` are missing const references to the pre-step data.